ruma-appservice-api/src/lib.rs

```diff
@@ -4,6 +4,69 @@
 
 #![warn(missing_debug_implementations, missing_docs)]
 
+use serde::{Deserialize, Serialize};
+
 pub mod event;
 pub mod query;
 pub mod thirdparty;
+
+/// A namespace defined by an application service.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Namespace {
```

Comment:
The new types should all be non-exhaustive by default, i.e. `#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]`, and had constructors.